bake/bake.go

```diff
@@ -0,0 +1,344 @@
+package bake
+
+import (
+	"context"
+	"io/ioutil"
+	"strings"
+
+	"github.com/moby/buildkit/session/auth/authprovider"
+	"github.com/pkg/errors"
+	"github.com/tonistiigi/buildx/build"
+)
+
+func ReadTargets(ctx context.Context, files, targets, overrides []string) (map[string]Target, error) {
+	var c Config
+	for _, f := range files {
+		cfg, err := ParseFile(f)
+		if err != nil {
+			return nil, err
+		}
+		c = mergeConfig(c, *cfg)
+	}
+	if err := c.setOverrides(overrides); err != nil {
+		return nil, err
+	}
+	m := map[string]Target{}
+	for _, n := range targets {
+		for _, n := range c.ResolveGroup(n) {
+			t, err := c.ResolveTarget(n)
+			if err != nil {
+				return nil, err
+			}
+			if t != nil {
+				m[n] = *t
+			}
+		}
+	}
+	return m, nil
+}
+
+func ParseFile(fn string) (*Config, error) {
+	dt, err := ioutil.ReadFile(fn)
+	if err != nil {
+		return nil, err
+	}
+
+	fnl := strings.ToLower(fn)
+	if strings.HasSuffix(fnl, ".yml") || strings.HasSuffix(fnl, ".yaml") {
+		return ParseCompose(dt)
+	}
+
+	if strings.HasSuffix(fnl, ".json") || strings.HasSuffix(fnl, ".hcl") {
+		return ParseHCL(dt)
+	}
+
+	cfg, err := ParseCompose(dt)
```

Comment:
Are there compose files without `.yml` or `.yaml` suffixes? Why not error out?

Reply:
I have no idea. Does compose force you to use `.yml` suffix currently, I think not.